kafka-producer-application/kafka/README.md

```diff
@@ -0,0 +1,166 @@
+# Kafka Producer
+
+An Apache Kafka® `Producer` is a client application that publishes (writes) events to a Kafka cluster.
+
+The KafkaProducer class implements multiple `send` methods, allowing the caller to provide `Callback` behavior once
+the event is sent to a Kafka topic. This tutorial will cover examples of both implementations.
+
+## Create a Producer and ProducerRecord
+
+There are required properties needed to create a Kafka Producer. At a minimum, the `Producer` needs to know:
+* How to find the Kafka broker(s).
+* How to serialize the key and value of events.
+
+```java annotate
+  final String bootstrapServers = "localhost:9092";
+  Properties properties = new Properties() {{
+      put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+      put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+      put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+      put(ProducerConfig.ACKS_CONFIG, "1");
+  }};
+  
+  Producer<String, String> producer = new KafkaProducer<>(properties);
+```
+
+For more about Kafka Producer Configurations, please [refer to the documentation](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html).
+
+To prepare an event to send, we create instances of `ProducerRecord` - which include the target topic, key and value, of the event
+to be sent to Kafka.
+
+```java annotate
+    final String key = "...";
+    final String value = "...";
+    final String topic = "...";
+    ProducerRecord<String, String> record = ProducerRecord<>(topic, key, value);
+```
+
+## Send, No Callback
+
+Given our `ProducerRecord`, let's send this record via our `Producer`. 
+
+```java annotate
+    producer.send(record);
```

Comment:
Maybe add here this method returns `Future<RecordMetadata>` and to get details of the produce result `Future.get()` my block if the request isn't completed.